Refactor fish scripts for fisher

rodrigobdz opened the pull request:
Fixes https://github.com/rodrigobdz/fish-apple-touchbar/issues/10.

See commit messages for further details.

jeremymeyers commented:
Looks good on glance, but not sure i know how to test/repicate at this point since the offending plugin is updated

rodrigobdz commented:
Tested changes by installing plugin from branch `refactor-fish-scripts-for-fisher`:

```sh
fisher install rodrigobdz/fish-apple-touchbar@refactor-fish-scripts-for-fisher
```

- [x] Fixed fisher's deprecation notice
- [x] Fixed fisher install and uninstall event triggers 

![Fixed fisher's deprecation notice](https://user-images.githubusercontent.com/14152377/109637945-b982fe00-7b4d-11eb-98f6-e6dc7d78a69b.png)
